File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res=[[1]]
        for i in range(numRows-1):
            a=[0]+res[-1]+[0]
            row=[]
            for j in range(len(res[-1])+1):
                row.append(a[j]+a[j+1])
            res.append(row)
        return res

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        if(numRows<=0){
            return res;
        }
        res.push_back({1});
        for(int i=1;i<numRows;i++){
            vector<int> tem={0};
            tem.insert(tem.end(),res.back().begin(), res.back().end());
            tem.push_back(0);
            vector<int> v;
            for (int j = 0; j < tem.size() - 1; j++){
                v.push_back(tem[j]+tem[j+1]);
            }
            res.push_back(v);

        }
        return res;

    }
};